content/en/contribute/code/core/deploy-on-eks.md

```diff
@@ -0,0 +1,132 @@
+
+## Starting and stopping (aka deleting)
+
+1. Login with `eks-aws-mfa-login` script in the [infra repo](https://github.com/medic/medic-infrastructure/tree/master/terraform/aws/dev/eks/access): `./eks-aws-mfa-login USERNAME  TOTP_HERE`
+2. Ensure you're using dev EKS cluster: `kubectl config use-context arn:aws:eks:eu-west-2:720541322708:cluster/dev-cht-eks`
+3. Add new `values.yaml` file - you can [copy this one](https://github.com/medic/medic-infrastructure/blob/master/terraform/aws/dev/cht-projects/alpha-dev-cht-deploy-values.yaml) and just change the `alpha-dev` values to `USERNAME-dev`
+4. Use `uuidgen` to fill in the `secret` in `values.yaml`
+5. Use a good passphrase (diceware!) to fill in `1password` in `values.yaml`
+6. Ensure you have the latest code of `cht-core` [repo](https://github.com/medic/cht-core): `git checkout master;git pull origin`
+7. Deploy!: `cd scripts/deploy;./cht-deploy -f PATH_TO/values.yaml`
+8. Delete it when you're done: `helm delete USERNAME-dev --namespace USERNAME-dev` 
+
+## References and Debugging
+
+More information on `cht-deploy` script is available in the [CHT Core GitHub repository](https://github.com/medic/cht-core/blob/master/scripts/deploy/README.md) which includes specifics of the `values.yaml` file and more details about the debugging utilities listed below.
+
+### Debugging
+
+A summary of the utilities in `cht-core/scripts/deploy` directory, assuming `mrjones-dev` namespace:
+
+* list all resources: `./troubleshooting/list-all-resources mrjones-dev` 
+* view logs, assuming `cht-couchdb-1` returned from prior command: `./troubleshooting/view-logs mrjones-dev cht-couchdb-1`
+* describe deployment, assuming `cht-couchdb-1` returned from 1st command: `./troubleshooting/describe-deployment mrjones-dev cht-couchdb-1`
+* list all deployments: `./troubleshooting/list-all-resources mrjones-dev`
+
+### Getting shell
+
+Sometimes you need to look at files and other key pieces of data that are not available with [the current](https://github.com/medic/cht-core/blob/master/scripts/deploy/troubleshooting/view-logs) `troubleshooting/view-logs` script.  In this case, getting an interactive shell on the pod can be helpful. 
+
+1. First, get a list pods for your namespace: `kubectl -n NAMESPACE get pods`
+2. After finding the pod you're interested, connect to the pod to get a shell: `kubectl -n NAMESPACE exec -it PODNAME/CONTAINERNAME -- /bin/bash`
+
+### `invalid apiVersion` Error
+
+If you get the error: 
+
+> exec plugin: invalid apiVersion "client.authentication.k8s.io/v1alpha1" when running `kubectl version`
+
+You are using an outdated version of kubernetes api `client.authentication.k8s.io` which is not supported by your `kubectl` client. This can sometimes happen if aws cli is an older version, ensure you are running at least version `2` of aws cli by running: `aws --version`
```

Comment:
Since the target environment is unknown I don't feel as sure of this
```suggestion
You might be using an version of kubernetes api `client.authentication.k8s.io` which is not supported by your `kubectl` client. This can sometimes happen in EKS clusters if aws cli is an older version, in most cases you need at least version `2` of aws cli. Check version by running: `aws --version` and note that version `2` *cannot* be installed through `pip` (See [Command Line](#command-line) section above for installation instructions)
```